src/transformState.js

```diff
@@ -4,8 +4,29 @@
  * @param {Object} state
  * @param {Object[]} actions
  */
+
 function transformState(state, actions) {
-  // write code here
+  for (const object of actions) {
+    for (const key in object) {
+      if (object[key] === 'clear') {
+        for (const key2 in state) {
+          delete state[key2];
+        }
+      }
+
+      if (object[key] === 'addProperties') {
+        Object.assign(state, object['extraData']);
+      }
+
+      if (object[key] === 'removeProperties') {
+        for (const item of object['keysToRemove']) {
+          delete state[item];
+        }
+      }
```

Comment:
Use [`switch case`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch) in cases where you constantly compare one value with others